elastic/kibana
# [Cloud Security] add url license locator

## Summary
resolves https://github.com/elastic/kibana/issues/153016

Since the License Management plugin is not available in ESS and serverless, this PR fetch a URL locator to the License Management plugin page safely, if the plugin is not available, instead having al link we will leave a message to the user. 

### How to test
1. Start ES with the trial license `yarn es snapshot --license=trial` and start Kibana `yarn start`.
2. Test1: Edit the `if (!subscriptionStatus.data)` condition in `x-pack/plugins/cloud_security_posture/public/components/cloud_posture_page.tsx` with `if (true)` and verify that the findings/dashboard pages contains valid link to the License management page.
3. Test2: `return undefined` instead checking if locator exists in `x-pack/plugins/cloud_security_posture/public/common/api/use_license_management_locator_api.tsx` and verify that instead of link there is an a message that says: "Contact your administrator to change your license."

### Link exists when the license plugin is available 
<img width="2489" alt="image" src="https://github.com/elastic/kibana/assets/90558359/2209afb7-b4c2-4241-8c37-f74091a1a8fe">


### Message when the license plugin is not available 
<img width="2499" alt="image" src="https://github.com/elastic/kibana/assets/90558359/bf637d06-12de-4cac-a4e3-4d21defd3044">
